### PR TITLE
added configureAwait(false) to all internal async calls. changed sync…

### PR DESCRIPTION
**Pull request by je-ewan**
… calls from .Result to  .ConfigureAwait(false).GetAwaiter().GetResult();

### CODE REVIEW

File: src/ZendeskApi.Client/Resources/RequestResource.cs [hunk 1]
```diff
@@ -28,7 +28,7 @@ public IResponse<Request> Get(long id)
 
         public IResponse<Request> Get(IEnumerable<TicketStatus> requestedStatuses)
         {
-            return GetAsync(requestedStatuses).Result;
+            return GetAsync(requestedStatuses).ConfigureAwait(false).GetAwaiter().GetResult();
```
Comment (EdNet) [inline src/ZendeskApi.Client/Resources/RequestResource.cs]: @kbrimble I am not too sure if that will avoid deadlock. I am testing that case currently in RM

Comment (EdNet) [inline src/ZendeskApi.Client/Resources/RequestResource.cs]: @kbrimble I am not too sure if that will avoid deadlock. For test I tried using GetAsync().ConfigureAwait(false).GetAwaiter().GetResult() and it went into deadlock.

Comment (kbrimble) [inline src/ZendeskApi.Client/Resources/RequestResource.cs]: @EdNet I've reworked all of the synchronous calls to not just call awaiters on async methods any more. See https://github.com/justeat/ZendeskApiClient/pull/57 for more info but BCL is having some issues so I can't get it out

**Comment by kbrimble** [comment]
I'm going to close this as we decided to make all of the synchronous methods not abuse async anymore.

**Pull request closed without merging**